### PR TITLE
issue #306 api: updates: before put response into cache, we should ta…

### PR DESCRIPTION
**Pull request by grinrag**
…ke into account what repos were passed in the input JSON.

### CODE REVIEW

File: webapp/updates.py [hunk 1]
```diff
@@ -347,12 +347,17 @@ def process_list(self, data):
             'update_list': {},
         }
 
+        # Get list of valid repository IDs based on input paramaters
+        available_repo_ids = self._process_repositories(data, response)
+
+        repo_ids_key = hashlib.md5('_'.join([str(id) for id in sorted(available_repo_ids)]).encode('utf-8')).hexdigest()
```
Comment (MichaelMraka) [inline webapp/updates.py]: _ is a valid character in repo label so something else (space or comma?)  would be better separator.

**Comment by MichaelMraka** [comment]
The very same issue apply also for releasever. We need to take it into account when building cache.

File: webapp/updates.py (hunk 1 above)
Comment (grinrag) [reply webapp/updates.py]: I'm joining repo IDs, so just numbers. Then it goes to the hash function.

**Comment by grinrag** [comment]
self._process_repositories() already takes into account `'repository_list'`, `'releasever'`, `'releasever'`

File: webapp/updates.py (hunk 1 above)
Comment (MichaelMraka) [reply webapp/updates.py]: I see, then it's ok.

**Comment by MichaelMraka** [comment]
Ah, I overlooked it, now it seems ok to me.

**Pull request merged**